### DISCUSSION TITLE
cant see historical data

### DISCUSSION THREAD

**Discussion by github731** [BottlecapDave/HomeAssistant-OctopusEnergy#567] [Q&A]
after some tweaking i managed to get this to work on the HA but I can only see data from the day I added it to HA, is there a method or setting somewhere in HA that can enable it to pull older data ?

**Comment by github731** [comment]
any help would be much appreciated :)

**Comment by BottlecapDave** [comment]
The integration by default doesn't pull in historic data. If you wish to do so, you can use the [refresh service](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/services.md#octopus_energyrefresh_previous_consumption_data) to pull in data. It's not really built for this, so your mileage may vary. This is only available for previous consumption sensors, and will be imported into the separate statistic sensors that are recommended for use with the energy dashboard. This is because HA entities are not built to have historic data pulled in.
